BackEnd/controllers/cupoOtorgado.controller.js

```diff
@@ -0,0 +1,238 @@
+      return res
+        .status(400)
+        .json({ message: "Ya cumplio su cuota diaria de reservas" });
+    }
+    //veriicar si cleinte posee una reseva activa para el dia de la clase
+    const [reservaActiva] = await pool.query(
+      `
+            SELECT count(*) as cantidad from cupo_otorgado
+            where fecha = CURDATE() and dniCliente = ? and estado = "reservado"`,
+      [dniCliente]
+    );
+    if (reservaActiva[0].cantidad > 0) {
+      return res
+        .status(400)
+        .json({ message: "Ya tiene una reserva activa para el dia de hoy" });
+    }
+
+    
+    const offset = -3; // Cambia por la diferencia horaria de tu zona (por ejemplo, -3 para Buenos Aires)
+
+    // Obtén la fecha actual en UTC
+    const fechaActualUTC = new Date();
+    
+    
+    const diferenciaHoraria = offset * 60 * 60 * 1000;
+    
+    
+    const fechaClase = new Date(fechaActualUTC.getTime() + diferenciaHoraria);
+    
+  
+    const fechaFormateada = fechaClase.toISOString().split("T")[0];
+
+    
+    const estado = "reservado";
+    
+    const [result] = await pool.query(
+      `
+            INSERT INTO cupo_otorgado (fecha, horaInicio, dniCliente, dniInstructor, estado) 
+            VALUES (?, ?, ?,?,?)`,
+      [fechaFormateada, horaInicio, dniCliente, dniInstructor, estado]
+    );
+    if (result.affectedRows === 0) {
+      return res.status(400).json({ message: "No se pudo otorgar el cupo" });
+    } else {
+      res.json({ message: "Cupo otorgado" });
+    }
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+export const obtenerCantCuposHoy = async (req, res) => {
+  try {
+    const [result] = await pool.query(`
+            SELECT horaInicio, count(*) as reservas from cupo_otorgado 
+            where fecha = CURDATE() and estado != 'cancelado'
+            group by horaInicio`);
+    if (!result || result.length === 0) {
+      return res.status(404).json({ message: "No hay cupos otorgados" });
+    } else {
+      res.status(200).json(result);
+    }
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+export const obtenerCuposOcupadosXidEsquema = async (req, res) => {
+  try {
+    const [result] = await pool.query(`
+            SELECT COUNT(*) AS reservas 
+            FROM cupo_otorgado c
+            WHERE c.fecha = CURDATE() 
+              AND (c.estado = 'reservado' OR c.estado = 'asistido')
+              AND c.horaInicio IN (SELECT e.horario FROM esquemacupos e WHERE e.idEsquema = ${req.params.idEsquema});
+
+            
+            `);
+
+    if (result.length === 0) {
+      return res.status(404).json({ message: "No hay cupos ocupados" });
+    } else {
+      res.json(result[0].reservas);
+    }
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+export const confirmarAsistencia = async (req, res) => {
+  try {
+    const { dniCliente } = req.params;
+    const horaAsistencia = new Date().toTimeString().split(" ")[0];
+
+    
+    const [response] = await pool.query(
```

Comment:
esta lib hace escaping de los parametros para prevenir sql injection ?

Reply:
Revisamos la documentacion de MySql2/promise (desde donde se importa pool) y si, lo hace solo si se lo pasa por parametro a cada campo. Si uno ejecutase un string directamente por ejemplo: const [cupoReservado] = await pool.query(
      
            SELECT count(*) as cantidad from cupo_otorgado
            where fecha = CURDATE() and dniCliente = ? and estado = "asistido",
      [dniCliente]
    );